internal/tools/changes/metadata.go

```diff
@@ -0,0 +1,235 @@
+package changes
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+// SchemaVersion defines the current JSON schema version for persistent data types (Change, Release, ...)
+const SchemaVersion = "1.0"
+
+// Metadata is a representation of the change metadata stored in a .changes directory.
+type Metadata struct {
+	ChangePath string     // ChangePath is the relative path from the current directory to .changes
+	Changes    []*Change  // Changes holds all pending change metadata in the .changes/next-release directory
+	Releases   []*Release // Releases contains all releases in the .changes/releases directory
+}
+
+// LoadMetadata loads the .changes directory at the given path.
+func LoadMetadata(path string) (*Metadata, error) {
+	changes, err := loadChanges(filepath.Join(path, "next-release"))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Metadata{
+		ChangePath: path,
+		Changes:    changes,
+	}, nil
+}
+
+// AddChange adds the given Change to the Metadata's Changes and saves the Change to the next-release directory.
+func (m *Metadata) AddChange(c *Change) error {
+	err := m.SaveChange(c)
+	if err != nil {
+		return err
+	}
+
+	m.Changes = append(m.Changes, c)
+	return nil
+}
+
+// AddChangesFromTemplate parses the given YAML template, adding the resulting Changes to Metadata's Changes and saving
+// the Changes to the next-release directory. AddChangesFromTemplate returns the created Changes.
+func (m *Metadata) AddChangesFromTemplate(template []byte) ([]*Change, error) {
+	changes, err := TemplateToChanges(template)
+	if err != nil {
+		return nil, err
+	}
+
+	return changes, m.AddChanges(changes)
+}
+
+// AddChanges adds the given Changes to Metadata's Changes and saves the Changes to the next-release directory.
+func (m *Metadata) AddChanges(changes []*Change) error {
+	for _, c := range changes {
+		err := m.AddChange(c)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// GetChangeById returns the pending Change with the given id.
+func (m *Metadata) GetChangeById(id string) (*Change, error) {
+	_, c, err := m.getChange(id)
+	return c, err
+}
+
+func (m *Metadata) getChange(id string) (int, *Change, error) {
+	for i, c := range m.Changes {
+		if c.ID == id {
+			return i, c, nil
+		}
+	}
+
+	return 0, nil, fmt.Errorf("couldn't find change with id %s", id)
+}
```

Comment:
Having `m.Changes` be loaded in sorted order allows the tool to have consistent change indices between runs. This is useful since `tool change ls` prints each change's index in `m.Changes`. That index can then be used as an argument to `tool change edit <index>` and `tool change rm <index>` instead of providing the full Change ID.